update docu for homebrew casks

aamotharald opened the pull request:
Hi colleagues,

this reflects the available ea and releases of sapmachine as homebrew casks in the Readme.

Maybe it also makes sense to mark released LTS (Long-Term-Support) releases as released LTS releases.

Happy to get your opinion there.

BR,

Harald

aamotharald committed:
update docu for homebrew casks

This reflects the available ea and releases of sapmachine as homebrew casks

parttimenerd commented:
Do we really want to mention releases that aren't supported anymore (like JDK 12)?

aamotharald commented:
Hi @parttimenerd ,

I agree with you that rather only LTS releases are mention-worthy. And old EA releases are also not really mention-worthy.
Or maybe List in the Readme:
Latest/Newest EA
LTS releases
optionally at the bottom a list of all releases so that people know what to type when making a brew uninstall (e.g to get rid of the SapMachine 11ea installation on their system that they forgot for a few years). But that might also be kind of self-explanatory..

aamotharald committed:
review comment

aamotharald committed:
markdown escaping ?

parttimenerd commented on README.md:
```diff
@@ -23,44 +23,24 @@ Tap it via:
+|--|--|--|
+| SapMachine 25 Early Access | `sapmachine25-ea-jdk` | `sapmachine25-ea-jre` |
+
+## Long Term Support (LTS)
```
Or call it supported versions (and include JDK 24) and don't include SapMachine 11, maybe.

aamotharald replied:
ah right SapMachine11 LTS ended Nov 2024 :-) .
Somehow I want to split LTS and non-LTS to make the LTS releases more prominent.

I also favor to only show the latest non-LTS. What do you think?

parttimenerd replied:
Maybe just name the LTS versions "__SapMachine 21 (LTS)__" and the other just "SapMachine 24"?

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
Update README.md

aamotharald committed:
added link to SapMachine Release Maintenance Page

aamotharald committed:
linebreak formatting

aamotharald committed:
typo fix

aamotharald committed:
formulation change

RealCLanger commented on README.md:
```diff
@@ -23,17 +24,27 @@ Tap it via:
-| SapMachine 17 Early Access | `sapmachine17-ea-jdk` | `sapmachine17-ea-jre` |
-| SapMachine 11 Released | `sapmachine11-jdk` | `sapmachine11-jre` |
-| SapMachine 11 Early Access | `sapmachine11-ea-jdk` | `sapmachine11-ea-jre` |
+| SapMachine 25 Early Access | `sapmachine25-ea-jdk` | `sapmachine25-ea-jre` |
```
```suggestion
| SapMachine 17 Early Access | `sapmachine17-ea-jdk` | `sapmachine17-ea-jre` |
| SapMachine 21 Early Access | `sapmachine21-ea-jdk` | `sapmachine21-ea-jre` |
| SapMachine 24 Early Access | `sapmachine24-ea-jdk` | `sapmachine24-ea-jre` |
| SapMachine 25 Early Access | `sapmachine25-ea-jdk` | `sapmachine25-ea-jre` |
```
Please keep SapMachine 17, 21 and 24 early access around as we publish pre-release versions for testing the upcoming update releases with that. Maybe this could be explicitly mentioned?

aamotharald replied:
taken over as suggested

RealCLanger commented on README.md:
```diff
@@ -23,17 +24,27 @@ Tap it via:
+
+## (Other) Available Versions
+
+Other Versions not listed in this Readme can be found by the pattern `sapmachine<XX>{-ea-}-<jdk|jre>` and can be found in the Casks subfolder. 
```
```suggestion
Other (outdated) versions not listed in this Readme can be found by the pattern `sapmachine<XX>{-ea-}-<jdk|jre>` and can be found in the Casks subfolder. 
```

RealCLanger reviewed (changes requested):
I think we should mention the released versions first, then the ea versions.

aamotharald committed:
outdated versions explicited

Co-authored-by: Christoph Langer <christoph.langer@sap.com>

aamotharald committed:
CL review comments adressed

I reordered from lowest to highest version like in the SapMachine Manitenance and Support page (to stay consistent there), moved the EA's below the Releases and added a denotion what Early Access means

aamotharald committed:
change awkward formulation

aamotharald committed:
removed artefact

RealCLanger commented:
Thanks for the cleanups @aamotharald. I'll merge this now.
